---
title: "RNA-SEQ"
author: "Diego Ramirez, Samuel QuiÃ±ones"
date: "11/4/2022"
output:
  html_document: default
  pdf_document: default
---



## ABSTRACT

DNA N-6-methyladenosine is a epigenetic modification in humans implicated in the progression and development of many types of cancer. However, the role of this modification in breast cancer is unclear; through bioinformatic analysis of RNA-seq data we found that the knockdown of the N6AMT1 gene result in the overexpression of genes involved with mithocondrial membranes, cytokine activity and intracellular signal transduction, in the other hand, the down-expressed genes were involved in metabolism and transport of nitrogenous bases. This results were obtained using a non-adjusted p-value, therefore this should be taken as a very general sight of this modification in breast cancer.

## INTRODUCTION

Breast cancer is one of the most common invasive cancers among women, being 25% of the total cases of cancer in women(1), a big amount of factors are involved in cancer development ranging from genetic predisposition, lifestyle to environment.

The molecular mechanisms of Breast Cancer progression are not well understood, but is known that an aberrant epigenetic pattern is an involved factor in this process. Particularly, the N6-Methyladenosilase is an epigenetic modification whose alteration could affect genic expression and splicing pattern significally, resulting in a regulation of genes involved in cell cycle(3), the role of this modification in the progression of glioblastoma, liver cancer and gastric cancer has been described(4)(5). However, the role of N6-Methyladenosilase in breast cancer is not well-understood.

The N6AMT1 gene produces a N-6 adenin-specific DNA methyltransferase responsible of this epigenetic modification. In this paper we conducted RNA-seq analysis to identify the related pathways with N6AMT1. To do so, we downloaded raw fastq data from ENA in wich a knockdown of N6AMT1 was performed.


## METHODOLOGY

A set of raw RNA-seq data was downloaded from European Nucleotide Archive (ENA) database, this fastq files consist in 6 samples of breast tumor (cell line MDA-MB-453), 3 of them treated with siRNA targeting the N6AMT1 gene and 3 of them were control; the GEO accession is GSE166582.
We downloaded the data from: https://www.ebi.ac.uk/ena/browser/view/PRJNA797275?show=reads

```{r, eval=F, echo=T}
## Downloading the data from ENA
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR136/020/SRR13681220/SRR13681220_1.fastq.gz 
#We did this for every single read in the dataset.
```

Then a quality control analysis was performed using the fastqc command.
```{r, eval=F, echo=T}
## Quality check 
module load fastqc/0.11.3
fastqc SRR13681220_1.fastq.gz #We did for every single read in the dataset.
module load multiqc/1.5
multiqc . # Make sure that all the output files of the fastqc command are in the same directory.
```

Our data failed the "Per-base sequence content module", but these are RNA-seq data so this bias in the firsts twelve nucleotides is expected.

![Figure 1](./PB-sequence_content.png)

Per Base Sequence Content
https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3AnalysisModules/4PerBaseSequenceContent.html

We also observed a little increment in the number of Duplication levels at the >10, but again, this is not a problem.

![Figure 2](./fastqc_sequence_duplication_levels_plot.png)

Duplications level
https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html
https://dnatech.genomecenter.ucdavis.edu/faqs/should-i-remove-pcr-duplicates-from-my-rna-seq-data/
https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4933-1

Once the quality control analysis was performed, a trimming of the adapter sequences was conduced using trimmomatic and TruSeq3-PE adapters, the adapters were downloanded from:  https://github.com/timflutre/trimmomatic/tree/master/adapters.

```{r, eval=F, echo=T}
module load trimmomatic/0.33 ;
trimmomatic PE -phred33 -basein ../SRR17618485_1.fastq.gz  -baseout ./SRR17618485_trmd.fastq.gz ILLUMINACLIP:./TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
```


- phred33 or -phred64 specifies the base quality encoding. If no quality encoding is specified,
it will be determined automatically (since version 0.32). The prior default was -phred64.

- ILLUMINNACLIP: This step is used to find and remove Illumina adapters

- SLIDINGWINDOW: Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold.

- MINLEN: This module removes reads that fall below the specified minimal length. If required, it should normally be after all other processing steps. 


Then we indexed the reference transcriptome (GRCh38) using the kallisto index command. The reference transcriptome was downloaded from https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.transcripts.fa.gz.


```{r, eval=F, echo=T}
module load kallisto/0.45.0 # Kallisto version loaded
kallisto index -i gencode_index gencode.v38.transcripts.fa.gz
```

- i, --index=STRING Filename for the kallisto index to be construted.

Once the index was created we mapped the reads using the kallisto quant command.

```{r, eval=F, echo=T}
kallisto quant -i ./gencode_index -o ./SRR13681220 ./SRR13681220_* # We ran this command for every RUN ID
```

- i, --index=STRING  Filename for the Kallisto index to be used for the quantification.

In order to Normalize and find differential expressed genes we used DEseq2 R package.

```{r, eval=F, echo=T}
library(tximport)
library(tidyverse) #stringr.

files <- file.path(list.dirs()[-c(1)],"abundance.h5") #Setting the path of the abundance files.
names(files) <- str_extract(files,"SRR\\d+") 
files

tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_ensgid-nover.csv",stringsAsFactors = F) #Loading ensembl annotation.

tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_genename-nover.csv",stringsAsFactors = F) #Loading ensembl annotation.

txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE) #Generating the matrix.
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE) #Generating the matrix.

#First we made a mistake mapping the Mus musculus genome instead the human genome and we had the following problem.

# -tximport error: all(txId == raw[[txIdCol]]) is not TRUE 
#  longer object length is not a multiple of shorter object length


samples <- read.csv("./SraRunTable(2).txt",stringsAsFactors = F, header = TRUE) #Obtained parsing.

samples <- column_to_rownames(samples, var = "Run") #Setting the rownames as the run ID of the samples.

library(DESeq2)

dds <- DESeqDataSetFromTximport(txi.kallisto,colData = samples,design = ~ transfected) #Constructing the dds object.

dds$transfected <- factor(dds$transfected, levels=c("control","si-N6AMT1")) #Indicating the groups to be compared.

dds <- DESeq(dds) #Performing DESeq analysis.

res <- results(dds) #Extracting a result table from a DESeq analysis giving base means across samples, log2 fold changes, standard errors, test statistics, p-values and adjusted p-values.

```

Once we have the dds object that contain the DEseq analysis, we created a volacano plot to visualize the overexpressed and the sub-expressed genes using the following code.

```{r, eval=F, echo=T}
## VOLCANO PLOT COMMANDS


de <- as.data.frame(res) #Adding a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative).


de$diffexpressed <- "NO"
# if log2Foldchange > 1 and pvalue < 0.01, set as "UP".
de$diffexpressed[de$log2FoldChange > 1 & de$padj < 0.05] <- "UP"
# if log2Foldchange < -1 and pvalue < 0.01, set as "DOWN".
de$diffexpressed[de$log2FoldChange < -1 & de$padj < 0.05] <- "DOWN"

library(ggrepel) # library that avoids the overlapping between text labels.

# Create a new column "names" to de, that will contain the name of a subset if genes differentially expressed (NA in case they are not).
de$names <- NA
# filter for a subset of interesting genes.
filter <- which(de$diffexpressed != "NO" & de$padj < 0.05 & (de$log2FoldChange > 1 | de$log2FoldChange < -1)) #<<
de$names[filter] <- rownames(de)[filter] #<<

# Generating and saving the plot.
png(file = paste0("./","volcano1-padj-res.png"),
    width = 800, height = 800) # Savig the plot in png format.

ggplot(data=de, aes(x=log2FoldChange, y=-log10(padj), col=diffexpressed, label=names)) + 
    geom_point() +
    scale_color_manual(values=c("blue", "black", "red")) + 
    theme_minimal() +
    geom_text_repel() + 
    xlim(-15,15)

dev.off() #This function closes the specified plot

```

Finally we performed a GO enrichment analysis using the output of the dds object to find over-representation of functions from Gene Ontology database.

```{r, eval=F, echo=T}

## GO ENRICHMENT ANALYSIS COMMANDS

library(ggrepel)
library(gprofiler2)

# For down-regulated genes

# subset results for genes of interest
resSig <- subset(res, pvalue < 0.05 & log2FoldChange < -1) 
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]

# define gene lists
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "hsapiens", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
               
library(forcats)
library(ggplot2)

outdir = "./" #Setting where the plot will be saved.
go <- as.data.frame(gostres$result)

#Creating and saving the plot.
png(file = paste0(outdir,"gost02-resSig.png"),
    width = 1800, height = 800) # saving the plot in png format.
go %>%
  dplyr::slice(1:15) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
  geom_bar(stat="identity", fill="#3A68AE", alpha=.6, width=.4) +
  coord_flip() +
  labs(x = "", y = "P-value") +
  theme_bw(base_size = 20)

dev.off() #This function closes the specified plot

# For up-regulated genes

# subset results for genes of interest
resSig <- subset(res, pvalue < 0.05 & log2FoldChange > 1) 
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]

# define gene lists
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "hsapiens", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
               
library(forcats)
library(ggplot2)

outdir = "./" #Setting where the plot will be saved.
go <- as.data.frame(gostres$result)

#Creating and saving the plot.
png(file = paste0(outdir,"gost03-resSig.png"),
    width = 1800, height = 800) # Saving the plot in png format.


go %>%
  dplyr::slice(1:15) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
  geom_bar(stat="identity", fill="#eb0c0c", alpha=.6, width=.4) +
  coord_flip() +
  labs(x = "", y = "P-value") +
  theme_bw(base_size = 20)

dev.off() #This function closes the specified plot

```

## Results 

We obtained that 16 genes were significantly up-regulated (adjusted p-value < 0.05) after si-N6AMT1 transfection (Figure 1). On the other hand, 11 genes were significantly down-regulated (adjusted p-value < 0.05) under the same conditions (Figure 1).

![Figure 3](./volcano1-padj-res.png)

To determine which functions are enriched within this subset of significantly expressed genes, we performed a GO enrichment. Unfortunately, we did not obtain enough results with previous genes subset, so we decided to use the normal p-value instead of the adjusted p-value in order to obtain a bigger subset of genes.


![Figure 4](./gost03-red2-resSig.png)
The functions enriched within the subset of up-regulated genes (p-value < 0.05) were (Figure 2) : MIB complex, TNF-alpha/NF-kappa B signaling complex, MICOS complex, TNF-alpha/NF-kappa B signaling complex 8, kinase maturation complex 2, SNARE complex, and MSP58-RINT1 complex. Whereas in the subset of down-regulated genes (p-value < 0.05) the functions enriched were: Pyrimidine catabolism, Purine catabolism, transport of vitamins, nucleosides and related molecules, and transport of nucleosides and free purine and pyrimidine bases across the plasma membrane.

![Figure 5](./gost02-resSig.png)}

## Discussion

For a better understanding of the GO analysis output, we looked for the GO characterization of the complexes previously mentioned in the CORUM database, which is a comprehensive resource of mammalian protein complexes (http://mips.helmholtz-muenchen.de/corum/). It turns out that, in the case of the up-regulated genes, the functions of the complexes are as follows: linkage of the inner and outer mitochondrial membranes (MIB complex), cytokine activity (TNF-alpha/NF-kappa B signaling complex (CHUK, BTRC, NFKB2, PPP6C, REL, CUL1, IKBKE, SAPS2, SAPS1, ANKRD28, RELA, SKP1) and complex 8), formation and maintenance of mitochondrial cristae (MICOS complex), intracellular signal transduction (Kinase maturation complex 2), vesicle fusion mediation (SNARE complex (RINT1, ZW10, p31)), and regulation of ribosome biogenesis (MSP58-RINT1 complex). Whereas, in the case of the down-regulated genes, the functions enriched were those related with nitrogenous base catabolism as well as transport of these and associated molecules such as nucleosides.
